1.0: libwacom is not LTO ready

When compiled with LTO optimisation linkiong of the test suite fails:

<pre>[tkloczko@barrel test]$ make
/bin/sh ../libtool  --tag=CC   --mode=link gcc  -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -flto -Os -no-install -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto -fuse-linker-plugin -o test-deprecated test-deprecated.o ../libwacom/libwacom.la  -ldl 
libtool: link: gcc -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -flto -Os -Wl,-z -Wl,relro -Wl,--as-needed -Wl,-z -Wl,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto -fuse-linker-plugin -o test-deprecated test-deprecated.o  ../libwacom/.libs/libwacom.so -lgudev-1.0 -lgobject-2.0 -lglib-2.0 -ldl -Wl,-rpath -Wl,/home/tkloczko/rpmbuild/BUILD/libwacom-0.99.901/libwacom/.libs
/usr/bin/ld: /tmp/test-deprecated.thrYTc.ltrans0.ltrans.o: in function `main&apos;:
/home/tkloczko/rpmbuild/BUILD/libwacom-0.99.901/test/test-deprecated.c:77: undefined reference to `libwacom_match_destroy@LIBWACOM_0.33&apos;
/usr/bin/ld: /home/tkloczko/rpmbuild/BUILD/libwacom-0.99.901/test/test-deprecated.c:78: undefined reference to `libwacom_match_new@LIBWACOM_0.33&apos;
/usr/bin/ld: /home/tkloczko/rpmbuild/BUILD/libwacom-0.99.901/test/test-deprecated.c:79: undefined reference to `libwacom_error_set@LIBWACOM_0.33&apos;
/usr/bin/ld: /home/tkloczko/rpmbuild/BUILD/libwacom-0.99.901/test/test-deprecated.c:80: undefined reference to `libwacom_stylus_destroy@LIBWACOM_0.33&apos;
/usr/bin/ld: /home/tkloczko/rpmbuild/BUILD/libwacom-0.99.901/test/test-deprecated.c:81: undefined reference to `libwacom_update_match@LIBWACOM_0.33&apos;
collect2: error: ld returned 1 exit status
make: *** [Makefile:664: test-deprecated] Error 1
make: Target &apos;all&apos; not remade because of errors.
</pre>

--- a/meson.build
+++ b/meson.build
@@ -543,13 +543,18 @@ if get_option('tests')
 		message('valgrind not found, disabling valgrind test suite')
 	endif
 
-	test_deprecated = executable('test-deprecated',
-				     'test/test-deprecated.c',
-				     dependencies: [dep_libwacom, dep_dl],
-				     include_directories: [includes_src],
-				     c_args: tests_cflags,
-				     install: false)
-	test('test-deprecated', test_deprecated, suite: ['all'])
+	# because of the tricks we use in calling the function and exposing
+	# it to begin with, LTO gets confused and this fails to link.
+	# Let's just disable it here.
+	if not get_option('b_lto')
+		test_deprecated = executable('test-deprecated',
+					     'test/test-deprecated.c',
+					     dependencies: [dep_libwacom, dep_dl],
+					     include_directories: [includes_src],
+					     c_args: tests_cflags,
+					     install: false)
+		test('test-deprecated', test_deprecated, suite: ['all'])
+	endif
 endif
 
 # This is a non-optional test

